st2api/st2api/controllers/v1/policies.py

```diff
@@ -0,0 +1,263 @@
+        except Exception as e:
+            LOG.exception(e.message)
+            abort(http_client.NOT_FOUND, e.message)
+            return
+
+        result = self.model.from_model(instance)
+        if result and self.include_reference:
+            resource_type = getattr(result, 'resource_type', None)
+            name = getattr(result, 'name', None)
+            result.ref = PolicyTypeReference(resource_type=resource_type, name=name).ref
+
+        LOG.debug('GET %s with ref_or_id=%s, client_result=%s',
+                  pecan.request.path, ref_or_id, result)
+
+        return result
+
+    def _get_all(self, **kwargs):
+        result = super(PolicyTypeController, self)._get_all(**kwargs)
+
+        if self.include_reference:
+            for item in result:
+                resource_type = getattr(item, 'resource_type', None)
+                name = getattr(item, 'name', None)
+                item.ref = PolicyTypeReference(resource_type=resource_type, name=name).ref
+
+        return result
+
+    def _get_by_ref_or_id(self, ref_or_id):
+        if PolicyTypeReference.is_reference(ref_or_id):
+            resource_db = self._get_by_ref(resource_ref=ref_or_id)
+        else:
+            resource_db = self._get_by_id(resource_id=ref_or_id)
+
+        if not resource_db:
+            msg = 'PolicyType with a reference of id "%s" not found.' % (ref_or_id)
+            raise Exception(msg)
+
+        return resource_db
+
+    def _get_by_id(self, resource_id):
+        try:
+            resource_db = self.access.get_by_id(resource_id)
+        except Exception:
+            resource_db = None
+
+        return resource_db
+
+    def _get_by_ref(self, resource_ref):
+        try:
+            ref = PolicyTypeReference.from_string_reference(ref=resource_ref)
+        except Exception:
+            return None
+
+        resource_db = self.access.query(name=ref.name, resource_type=ref.resource_type).first()
+        return resource_db
+
+    def _get_filters(self, **kwargs):
+        filters = copy.deepcopy(kwargs)
+        ref = filters.get('ref', None)
+
+        if ref:
+            try:
+                ref_obj = PolicyTypeReference.from_string_reference(ref=ref)
+            except InvalidReferenceError:
+                raise
+
+            filters['name'] = ref_obj.name
+            filters['resource_type'] = ref_obj.resource_type
+            del filters['ref']
+
+        return filters
+
+
+class PolicyController(resource.ContentPackResourceController):
+    model = PolicyAPI
+    access = Policy
+
+    supported_filters = {
+        'pack': 'pack'
+    }
+
+    query_options = {
+        'sort': ['pack', 'name']
+    }
+
+    @jsexpose(body_cls=PolicyAPI, status_code=http_client.CREATED)
+    def post(self, instance):
+        """
+            Create a new policy.
+            Handles requests:
+                POST /policies/
+        """
+        op = 'POST /policies/'
+
+        if not hasattr(instance, 'enabled'):
```

Comment:
Yes.  I just copied this from https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/v1/actions.py#L80.